### PR TITLE
add agent config

### PR DESCRIPTION
**Pull request by bziser**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/common/constants.py [hunk 1]
```diff
@@ -279,6 +286,7 @@ class FileType(str, Enum):
     LISTS_DIR,
     JOBS_DIR,
     WIZARDS_DIR,
+    AGENT_CONFIG_DIR,
```
Comment (yaakovi) [inline demisto_sdk/commands/common/constants.py:289]: Do we support uploading agent config files directly?

File: demisto_sdk/commands/common/constants.py [hunk 2]
```diff
@@ -1107,6 +1116,7 @@ class PB_Status:
     'Scripts': SCRIPT_PREFIX,
     'ModelingRules': MODELING_RULE_PREFIX,
     'ParsingRules': PARSING_RULE_PREFIX,
+    'AgentConfigs': AGENT_CONFIG_PREFIX
```
Comment (yaakovi) [inline demisto_sdk/commands/common/constants.py:1119]:
```suggestion
    'AgentConfigs': AGENT_CONFIG_PREFIX,
```

File: TestSuite/pack.py [hunk 3]
```diff
@@ -650,3 +656,8 @@ def create_trigger(self, name, content: dict = {}) -> Trigger:
         trigger = Trigger(name, self._triggers_path, content)
         self.triggers.append(trigger)
         return trigger
+
+    def create_agent_config(self, name, json_content: dict = {}, yaml_content: dict = {}) -> AgentConfig:
```
Comment (yaakovi) [inline TestSuite/pack.py:660]: https://florimond.dev/blog/articles/2018/08/python-mutable-defaults-are-the-source-of-all-evil/
```suggestion
    def create_agent_config(self, name, json_content: dict = None, yaml_content: dict = None) -> AgentConfig:
```

File: demisto_sdk/commands/common/content/tests/objects/pack_objects/agentconfig/agent_config_test.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
+from demisto_sdk.commands.common.content.objects.pack_objects import \
+    AgentConfig
+from demisto_sdk.commands.common.content.objects_factory import \
+    path_to_pack_object
+
+
+def get_agent_config(pack, name):
```
Comment (yaakovi) [inline demisto_sdk/commands/common/content/tests/objects/pack_objects/agentconfig/agent_config_test.py:7]: Why do you need a function for it?

File: demisto_sdk/commands/common/tests/update_id_set_test.py [hunk 5]
```diff
@@ -3070,6 +3071,44 @@ def test_process_triggers(mocker, capsys, pack):
...
+        assert 'name' in agent_config_result.keys()
+        assert 'display_name' in agent_config_result.keys()
+        assert 'file_path' in agent_config_result.keys()
+        assert 'pack' in agent_config_result.keys()
+
+        assert agent_config_result['name'] == agent_config._file_path.parts[-1].split('.')[0]
+        assert agent_config_result['file_path'] == agent_config.path
+        assert agent_config_result['pack'] == pack.name
+
+        assert f'adding {agent_config._file_path} to id_set' in captured.out
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tests/update_id_set_test.py:3109]: nice

File: demisto_sdk/commands/common/tools.py [hunk 6]
```diff
@@ -1378,6 +1380,9 @@ def find_type_by_path(path: Union[str, Path] = '') -> Optional[FileType]:
             # Packs/myPack/Scripts/myScript/myScript.yml
             return FileType.SCRIPT
 
+        elif 'AgentConfigs' in path.parts:
```
Comment (yaakovi) [inline demisto_sdk/commands/common/tools.py:1383]:
```suggestion
        elif AGENT_CONFIG_DIR in path.parts:
```

File: demisto_sdk/commands/common/update_id_set.py [hunk 7]
```diff
@@ -1298,6 +1299,23 @@ def get_correlation_rule_data(path: str, packs: Dict[str, Dict] = None):
...
 
 
+def get_agent_config_data(path: str, packs: Dict[str, Dict] = None):
+    json_data = get_json(path)
+
+    id_ = json_data.get('content_global_id')
+    name = json_data.get('name')
+    display_name = get_display_name(path, json_data)
+    fromversion = json_data.get('fromVersion')
+    toversion = json_data.get('toVersion')
```
Comment (yaakovi) [inline demisto_sdk/commands/common/update_id_set.py:1309]:
```suggestion
    fromversion = json_data.get('from_xdr_version')
    toversion = json_data.get('to_xdr_version')
```

File: demisto_sdk/commands/create_id_set/tests/create_id_set_test.py [hunk 8]
```diff
@@ -177,10 +177,10 @@ def test_create_id_set_flow(repo, mocker):
 
     id_set_content = repo.id_set.read_json_as_dict()
     assert not IsEqualFunctions.is_dicts_equal(id_set_content, {})
-    assert set(id_set_content.keys()) == set(ID_SET_ENTITIES + ['Packs'])
+    assert set(id_set_content.keys()) == set(ID_SET_ENTITIES + ['Packs'])  # the test is not checking the right entities
```
Comment (yaakovi) [inline demisto_sdk/commands/create_id_set/tests/create_id_set_test.py:180]: ?

File: demisto_sdk/commands/unify/agent_config_unifier.py [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+                directory_name = optional_dir_name
+
+        if not directory_name:
+            print_error(UNSUPPORTED_INPUT_ERR_MSG)
+
+        if dir_name:
+            self.dir_name = dir_name
+
+        self.package_path = input
+        self.package_path = self.package_path.rstrip(os.sep)
```
Comment (yaakovi) [inline demisto_sdk/commands/unify/agent_config_unifier.py:41]:
```suggestion
        self.package_path = input.rstrip(os.sep)
```

File: demisto_sdk/commands/unify/agent_config_unifier.py [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+        input = os.path.abspath(input)
+        if not os.path.isdir(input):
+            print_error(UNSUPPORTED_INPUT_ERR_MSG)
+            sys.exit(1)
+        for optional_dir_name in DIR_TO_PREFIX:
+            if optional_dir_name in input:
+                directory_name = optional_dir_name
+
+        if not directory_name:
+            print_error(UNSUPPORTED_INPUT_ERR_MSG)
```
Comment (yaakovi) [inline demisto_sdk/commands/unify/agent_config_unifier.py:35]: `directory_name` is not used

File: demisto_sdk/commands/unify/agent_config_unifier.py [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+        self.package_path = self.package_path.rstrip(os.sep)
+
+        _, self.yml_path = get_yml_paths_in_dir(self.package_path, Errors.no_yml_file(self.package_path))
+
+        self.dest_path = os.path.abspath(output) if output else None
+
+    def unify(self):
+        click.echo(f'Unifying {self.package_path}...')
+        self._set_dest_path()
+        output_data = {}
```
Comment (yaakovi) [inline demisto_sdk/commands/unify/agent_config_unifier.py:50]:
```suggestion
```

**Comment by yaakovi** [comment]
Also, ignore the new file types in the `validate` command for now.
See https://github.com/demisto/demisto-sdk/blob/master/demisto_sdk/commands/validate/validate_manager.py#L205

File: TestSuite/pack.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (bziser) [reply TestSuite/pack.py:660]: 😮

File: demisto_sdk/commands/common/content/tests/objects/pack_objects/agentconfig/agent_config_test.py (hunk 4 above)
Comment (bziser) [reply demisto_sdk/commands/common/content/tests/objects/pack_objects/agentconfig/agent_config_test.py:7]: No need :)

File: TestSuite/agent_config.py [hunk 12]
```diff
@@ -5,7 +5,7 @@
 
 
 class AgentConfig(JSONBased, YAML):
```
[the remark this answers is not in the document]
Comment (bziser) [reply TestSuite/agent_config.py:7]: removed it

**Pull request merged**